port/unix/omrsock.c

```diff
@@ -952,6 +952,9 @@ omrsock_poll(struct OMRPortLibrary *portLibrary, omrsock_pollfd_t fds, uint32_t
 	} else {
 		/* Dynamically allocate if nfds more than 8. */
 		pfds = portLibrary->mem_allocate_memory(portLibrary, nfds * sizeof(struct pollfd), OMR_GET_CALLSITE(), OMRMEM_CATEGORY_PORT_LIBRARY);
+		if (NULL == pfds) {
+			portLibrary->error_set_last_error(portLibrary, errno, OMRPORT_ERROR_SYSTEMFULL);
+		}
```

Comment:
If this allocation fails, a segmentation fault will occur on line 959.

Reply:
Thanks for catching this Keith, I opened a fix for this in PR: https://github.com/eclipse/omr/pull/5495.